### ISSUE TITLE
Runtime Performance

### ISSUE THREAD

**Issue by ahundt** [gaudetcj/DeepQuaternionNetworks#3]
When I try running the mnist example, runtime performance for training looks like it will be about 2 hours on CPU for a single epoch. For comparison Keras' mnist example takes 2 minutes on CPU for each epoch. Is this 60x difference expected?

**Comment by gaudetcj** [comment]
Yes the parameters from the mnist tests are those used by Hinton in his
comparison to CNNs against Capsule Nets.
I believe it is quite a bit bigger than the mnist from the keras examples.
I have never run any of this code on CPU but it should not be much
different from running a regular CNN on CPU unless you are using the
quaternion batch normalization which still increase time by a good bit.
